### PR TITLE
HPCC-20296 Document __OPTION__(legacy[(true/false)])

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>
@ghalliday Please review. PDF sent via email

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by JamesDeFabia** [comment]
@AttilaVamos Can you look into this smoketest?

**Comment by AttilaVamos** [comment]
@JamesDeFabia It is a known issue and there is a PR( https://github.com/hpcc-systems/HPCC-Platform/pull/11576 )  to fix it.

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: docs/EN_US/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml [hunk 1]
```diff
@@ -294,6 +294,16 @@
           <entry>Optional. Specifies records are created on every
           node.</entry>
         </row>
+
+        <row>
+          <entry><emphasis role="bold">__OPTION__(LEGACY)</emphasis></entry>
+
+          <entry>Optional. Specifies that dataset contains deprecated record
```
Comment (ghalliday) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml]: I'm not sure this is quite right.  More like:

Specifies that the dataset should be read using the legacy mechanism, which does not support remote field projection.  This may be required if the dataset contains a combination of alien data types and virtual fields.

File: docs/EN_US/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml [hunk 2]
```diff
@@ -341,6 +351,14 @@
   <para>The ninth form is used to create a DATASET using a TRANSFORM function
   (see <emphasis role="bold">DATASET from TRANSFORM</emphasis> below)</para>
 
+  <para>The __OPTION__(LEGACY) option can be used if your dataset definition
```
Comment (ghalliday) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml]: Why is this here?  Most of the other optional options e.g. distributed, local, do not have any text in this section.  This section is describing the different format of the DATASET command.

File: docs/EN_US/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml [hunk 3]
```diff
@@ -341,6 +351,14 @@
...
+  contains deprecated record formats and virtual fields. Use this option
+  temporarily until you can remove alien data types from the record
+  definition. If this is needed, you will receive a compiler error
+  message.</para>
+
+  <programlisting>d := DATASET('imgfile',{rawLayout x,unsigned8 _fpos{virtual(fileposition)} },FLAT, __OPTION__(LEGACY)); </programlisting>
```
Comment (ghalliday) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml]: None of the other attributes have an example here - it looks out of place.

**Review by ghalliday — CHANGES_REQUESTED** [review]

**Comment by ghalliday** [comment]
@JamesDeFabia please review

**Comment by richardkchapman** [comment]
@JamesDeFabia Bump

**Comment by JamesDeFabia** [comment]
Closing this one, need to start over with a better approach
